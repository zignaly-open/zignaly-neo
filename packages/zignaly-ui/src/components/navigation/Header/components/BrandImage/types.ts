import { ReactComponent as ZignalyIsotype } from "../../../../../assets/images/zignaly-isotype.svg";
import { ReactComponent as ZignalyLogotype } from "../../../../../assets/images/zignaly-logotype.svg";

// Icons
export const iconsByType = {
  isotype: ZignalyIsotype,
  logotype: ZignalyLogotype,
};

export type LogoProps = {
  type: keyof typeof iconsByType;
  id?: string;
  width?: string;
  height?: string;
};
