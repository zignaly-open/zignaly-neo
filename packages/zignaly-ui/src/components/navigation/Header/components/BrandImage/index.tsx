import React from "react";
import { LogoProps, iconsByType } from "./types";

// TODO: fix types to logo and iso and add default variables
export default function BrandImage({ type, width, height, id }: LogoProps) {
  const Icon = iconsByType[type];
  return Icon ? <Icon width={width} height={height} id={id} /> : null;
}
