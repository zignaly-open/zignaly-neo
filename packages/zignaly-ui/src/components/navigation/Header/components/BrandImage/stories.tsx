// Dependencies
import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import BrandImage from "./";
import { iconsByType } from "./types";

export default {
  title: "Navigation/Header/components/BrandImage",
  component: BrandImage,
  argTypes: {
    type: {
      control: {
        type: "select",
      },
      options: Object.keys(iconsByType),
    },
  },
} as ComponentMeta<typeof BrandImage>;

const Template: ComponentStory<typeof BrandImage> = (args) => <BrandImage {...args} />;

export const Logotype = Template.bind({});
Logotype.args = {
  type: "logotype",
  width: "140px",
  height: "68px",
};

export const Isotype = Template.bind({});
Isotype.args = {
  type: "isotype",
  width: "40px",
  height: "40px",
};
