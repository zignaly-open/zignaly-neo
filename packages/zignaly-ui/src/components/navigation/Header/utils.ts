// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
