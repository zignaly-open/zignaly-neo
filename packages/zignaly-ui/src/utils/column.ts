import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
