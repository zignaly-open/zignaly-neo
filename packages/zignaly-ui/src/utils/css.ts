import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
