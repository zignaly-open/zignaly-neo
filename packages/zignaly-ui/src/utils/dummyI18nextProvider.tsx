import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
