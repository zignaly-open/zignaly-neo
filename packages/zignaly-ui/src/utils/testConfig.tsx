import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
