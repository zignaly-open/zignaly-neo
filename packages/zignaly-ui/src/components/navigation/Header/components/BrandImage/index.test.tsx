import React from "react";
import BrandImage from ".";
import { renderWithProvidersUi } from "../../../../../utils/testConfig";

describe("components/navigation/Header/components/BrandImage", () => {
  it("renders with isotype prop without crashing", async () => {
    const { container } = renderWithProvidersUi(<BrandImage type={"isotype"} />);

    const imageStyles = getComputedStyle(container);
    expect(container).toMatchSnapshot();
    expect(imageStyles).toMatchSnapshot();
  });

  it("renders with logotype prop without crashing", async () => {
    const { container } = renderWithProvidersUi(<BrandImage type={"logotype"} />);

    const imageStyles = getComputedStyle(container);
    expect(container).toMatchSnapshot();
    expect(imageStyles).toMatchSnapshot();
  });
});
