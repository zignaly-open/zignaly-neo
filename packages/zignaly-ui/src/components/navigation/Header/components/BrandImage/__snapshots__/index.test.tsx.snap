// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/navigation/Header/components/BrandImage renders with isotype prop without crashing 1`] = `
<div>
  <svg />
</div>
`;

exports[`components/navigation/Header/components/BrandImage renders with isotype prop without crashing 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "visibility",
  "_importants": {
    "display": "",
    "visibility": undefined,
  },
  "_length": 2,
  "_onChange": [Function],
  "_values": {
    "display": "block",
    "visibility": "visible",
  },
}
`;

exports[`components/navigation/Header/components/BrandImage renders with logotype prop without crashing 1`] = `
<div>
  <svg />
</div>
`;

exports[`components/navigation/Header/components/BrandImage renders with logotype prop without crashing 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "visibility",
  "_importants": {
    "display": "",
    "visibility": undefined,
  },
  "_length": 2,
  "_onChange": [Function],
  "_values": {
    "display": "block",
    "visibility": "visible",
  },
}
`;
